src/grid_search.jl

```diff
@@ -271,15 +272,28 @@ function _fit!(self::BaseSearchCV, X, y, parameter_iterable)
     base_estimator = clone(self.estimator)
 
     @assert self.n_jobs == 1 "TODO: support n_jobs > 1"
-    out = vcat(Any[[_fit_and_score(clone(base_estimator), X, y, self.scorer_,
+    out = []
+    if nprocs() == 1
+        out = vcat(Any[[_fit_and_score(clone(base_estimator), X, y, self.scorer_,
                                    train, test, self.verbose,
                                    kwargify(parameters),
                                    kwargify(self.fit_params),
                                    return_parameters=true,
                                    error_score=self.error_score)
                     for (train, test) in cv]
                    for parameters in parameter_iterable]...)
-
+    else
+        out = @sync @parallel (vcat) for parameters in parameter_iterable
+           [_fit_and_score(clone(base_estimator), X, y, self.scorer_,
```

Comment:
To reduce code duplication, could you please factor out the comprehension body into a separate local function? I.e.

```julia
    do_fit(train, test) = _fit_and_score(...)
    ...
        out = ...[do_fit(train, test) for ...]
```